### PR TITLE
Fix fileinfo for arweave file

### PR DESCRIPTION
**Pull request by mariacarmina**
Fixes # .

Changes proposed in this PR:

- Added check for file path validation inside UrlFile class.
- Fix fileinfo for arweave file.
- Updated tests.

### CODE REVIEW

File: ocean_provider/file_types/definitions.py [hunk 1]
```diff
@@ -72,7 +72,10 @@ def check_details(self, with_checksum=False):
                 if result:
                     status_code = result.status_code
                     headers = copy.deepcopy(result.headers)
-                    files_url = copy.deepcopy(result.url)
+                    if "arweave" in result.url:
```
Comment (alexcos20) [inline ocean_provider/file_types/definitions.py:75]: what if arweave gateway url is  "http://gateway.ar.example" ?
honestly, I would move this to Arweave class  (overwrite check_details) and do it for every arweave file

File: ocean_provider/file_types/definitions.py [hunk 2]
```diff
@@ -72,7 +72,10 @@ def check_details(self, with_checksum=False):
...
+                        files_url = ""
```
Comment (alexcos20) [inline ocean_provider/file_types/definitions.py:76]: can we get at least Content-size ?

**Review by alexcos20 — CHANGES_REQUESTED** [review]

File: ocean_provider/file_types/definitions.py (hunk 2 above)
Comment (mariacarmina) [reply ocean_provider/file_types/definitions.py:76]: I return content-length and content-type, these come from headers.

File: ocean_provider/file_types/definitions.py (hunk 1 above)
Comment (mariacarmina) [reply ocean_provider/file_types/definitions.py:75]: Sure, thanks for pointing out.

File: ocean_provider/file_types/definitions.py (hunk 2 above)
Comment (mariacarmina) [reply ocean_provider/file_types/definitions.py]: I assume you referred to Content-Length

**Review by alexcos20 — APPROVED** [review]
lgtm

**Pull request merged**